# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1744834449 -7200
# Parent  9d7fb70a951e644586806160acfe3e503a17825b
NNN8 - Don't log Cr.NS_ERROR_ILLEGAL_VALUE in MessageSend._doFcc2().

diff --git a/mailnews/compose/src/MessageSend.sys.mjs b/mailnews/compose/src/MessageSend.sys.mjs
--- a/mailnews/compose/src/MessageSend.sys.mjs
+++ b/mailnews/compose/src/MessageSend.sys.mjs
@@ -1093,9 +1093,15 @@ export class MessageSend {
         }
       } catch (e) {
         // Ignore the return value of onStopCopy. Non-zero nsresult will throw
         // when going through XPConnect. In this case, we don't care about it.
-        console.warn("onStopCopy failed", e);
+        // Do not log NS_ERROR_ILLEGAL_VALUE (0x80070057), it comes from here:
+        // https://searchfox.org/comm-central/rev/1dcff0c305e7a4ec5a91567d121724baf0d8e741/mailnews/compose/src/nsMsgCompose.cpp#3250
+        // Nothing else in `nsMsgComposeSendListener::OnStopCopy()` sets `rv`.
+        if (e.result != Cr.NS_ERROR_ILLEGAL_VALUE) {
+          console.warn(
+            `onStopCopy failed with 0x${e.result.toString(16)}\n${e.stack}`
+          );
       }
       this._cleanup();
     });
   }
